dipy/reconst/ivim.py

```diff
@@ -0,0 +1,541 @@
+            to leastsq.
+            default : options={'gtol': 1e-15, 'ftol': 1e-15, 'eps': 1e-15,
+                      'maxiter': 1000}
+
+        References
+        ----------
+        .. [1] Le Bihan, Denis, et al. "Separation of diffusion and perfusion
+               in intravoxel incoherent motion MR imaging." Radiology 168.2
+               (1988): 497-505.
+        .. [2] Federau, Christian, et al. "Quantitative measurement of brain
+               perfusion with intravoxel incoherent motion MR imaging."
+               Radiology 265.3 (2012): 874-881.
+        """
+        if not np.any(gtab.b0s_mask):
+            e_s = "No measured signal at bvalue == 0."
+            e_s += "The IVIM model requires signal measured at 0 bvalue"
+            raise ValueError(e_s)
+
+        ReconstModel.__init__(self, gtab)
+        self.split_b_D = split_b_D
+        self.split_b_S0 = split_b_S0
+        self.bounds = bounds
+        self.tol = tol
+        self.f_threshold = f_threshold
+        self.options = options
+        self.x_scale = x_scale
+
+        if SCIPY_LESS_0_17 and self.bounds is not None:
+            e_s = "Scipy versions less than 0.17 do not support "
+            e_s += "bounds. Please update to Scipy 0.17 to use bounds"
+            raise ValueError(e_s)
+        elif self.bounds is None:
+            self.bounds = ((0., 0., 0., 0.), (np.inf, 1., 1., 1.))
+        else:
+            self.bounds = bounds
+
+    @multi_voxel_fit
+    def fit(self, data, mask=None):
+        """ Fit method of the Ivim model class.
+
+        The fitting takes place in the following steps: Linear fitting for D
+        (bvals > 200) and store S0_prime. Another linear fit for S0 (bvals <
+        200).Estimate f using 1 - S0_prime/S0. Use least squares to fit D_star
+        and f.
+
+        We do a final fitting of all four parameters and select the set of
+        parameters which make sense physically. The criteria for selecting a
+        particular set of parameters is checking the perfusion fraction.
+        If the fraction is more than `f_threshold` (default: 25%), we will
+        reject the solution obtained from least squares fitting and consider
+        only the linear fit.
+
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel. A multi voxel decorator
+            will be applied to this fit method to scale it and apply it
+            to multiple voxels.
+
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[:-1]
+
+        Returns
+        -------
+        IvimFit object
+        """
+        # Get S0_prime and D.
+        S0_prime, D = self.estimate_linear_fit(data, self.split_b_D, less_than=False)
+        # Get S0 and D_star_prime.
+        S0, D_star_prime = self.estimate_linear_fit(data, self.split_b_S0,
```

Comment:
Do you define `S0_prime` or `D_star_prime` anywhere? It probably would be good to help explain what is going on here. If not I think here is fine because they're kind of intermediate variables not truly related to the model. But `S0_prime` is just the intercept of the diffusion signal, `D_star_prime` is the linear estimate of `D_star`.